token/cli/src/main.rs

```diff
@@ -2044,6 +2051,7 @@ fn main() {
             let fund_recipient = matches.is_present("fund_recipient");
             let allow_unfunded_recipient = matches.is_present("allow_empty_recipient")
                 || matches.is_present("allow_unfunded_recipient");
+            no_wait = matches.is_present("no_wait");
```

Comment:
@CriesofCarrots as discussed on dm lightly, I restricted `no_wait` only to `spl-token transfer`.